arch/sim/src/sim/sim_netdriver.c

```diff
@@ -334,22 +358,21 @@ int sim_netdriver_init(void)
                   netdriver_txdone_interrupt,
                   netdriver_rxready_interrupt);
 
-      /* Update the buffer size */
-
-      pktsize = dev->d_pktsize ? dev->d_pktsize :
-                (MAX_NETDEV_PKTSIZE + CONFIG_NET_GUARDSIZE);
-
       /* Allocate packet buffer */
 
-      pktbuf = kmm_malloc(pktsize);
-      if (pktbuf == NULL)
+#ifdef SIM_NETDEV_IOB_OFFLOAD
+      dev->d_buf = NULL;
+#else
```

Comment:
```suggestion
#ifndef SIM_NETDEV_IOB_OFFLOAD
```

Reply:
Why?

Reply:
g_sim_dev is global variable d_buf is already zero